agent/mapcall/src/main/java/randoop/instrument/MapCallsAgent.java

```diff
@@ -1,322 +1,195 @@
+    if (agentArgs != null) { // if there are any arguments, parse them
+      Options options = new Options(MapCallsAgent.class);
+      String[] target_args = options.parse_or_usage(agentArgs);
+      if (target_args.length > 0) {
+        System.err.printf("Unexpected agent arguments %s%n", Arrays.toString(target_args));
+        System.exit(1);
       }
     }
 
-    // Read the map file
-    if (map_calls != null) {
-      CallReplacementTransformer instrument = (CallReplacementTransformer) transformer;
-      instrument.read_map_file(map_calls);
-      instrument.add_map_file_shutdown_hook();
+    if (debug && !debug_directory.isEmpty()) {
+      debugPath = debugPath.resolve(debug_directory);
+      if (!Files.exists(debugPath)) {
+        Files.createDirectory(debugPath);
+      }
     }
 
-    // Instrument transformer = new Instrument();
-    inst.addTransformer((ClassFileTransformer) transformer);
-  }
+    // Load package prefixes from the resource file in the jar for default package exclusions
+    Set<String> excludedPackagePrefixes = new LinkedHashSet<>();
 
-  /**
-   * Reads purity file. Each line should contain exactly one method. Care must be taken to supply
-   * the correct format.
-   *
-   * <p>From the Sun JDK API:
-   *
-   * <p>"The string is formatted as the method access modifiers, if any, followed by the method
-   * return type, followed by a space, followed by the class declaring the method, followed by a
-   * period, followed by the method name, followed by a parenthesized, comma-separated list of the
-   * method's formal parameter types. If the method throws checked exceptions, the parameter list is
-   * followed by a space, followed by the word throws followed by a comma-separated list of the
-   * thrown exception types. For example:
-   *
-   * <p>public boolean java.lang.Object.equals(java.lang.Object)
-   *
-   * <p>The access modifiers are placed in canonical order as specified by "The Java Language
-   * Specification". This is public, protected or private first, and then other modifiers in the
-   * following order: abstract, static, final, synchronized native."
-   */
-  private static HashSet<String> readPurityFile(File purityFileName, File pathLoc)
-      throws IOException {
-    HashSet<String> pureMethods = new LinkedHashSet<>();
-
-    BufferedReader reader =
-        new BufferedReader(new FileReader(new File(pathLoc, purityFileName.getPath())));
-
-    if (true) System.out.printf("Reading '%s' for pure methods %n", purityFileName);
-
-    for (String line = reader.readLine(); line != null; line = reader.readLine()) {
-      pureMethods.add(line.trim());
+    String exclusionFileName = "/default-load-exclusions.txt";
+    InputStream inputStream = MapCallsAgent.class.getResourceAsStream(exclusionFileName);
+    if (inputStream == null) {
+      System.err.println("Unable to find default package exclusion file. Please report.");
+      System.exit(1);
```

Comment:
I understand, but it's still surprising to a reader to see that BugInRandoopException is used in other parts of the code but System.exit is called here.  Someone reading this code won't appreciate the difference, and I don't see documentation to help a reader understand when Randoop should use BugInRandoopException and when it should use System.exit().  This isn't a huge deal, but it did surprise me when reading it.